crates/biome_dependency_graph/src/module_visitor.rs

```diff
@@ -0,0 +1,385 @@
+                                }
+                            }
+                        },
+                        AnyJsObjectBindingPatternMember::JsBogusBinding(_) | AnyJsObjectBindingPatternMember::JsMetavariable(_) => {}
+                    }
+                }
+            }
+        }
+
+        Some(())
+    }
+
+    fn visit_identifier_binding(
+        &mut self,
+        binding: &JsIdentifierBinding,
+        jsdoc_comment: Option<String>,
+    ) -> Option<()> {
+        let name = binding.name_token().ok()?.token_text_trimmed();
+        self.register_export_with_name_and_jsdoc_comment(name, jsdoc_comment)
+    }
+
+    fn register_export_with_name_and_jsdoc_comment(
+        &mut self,
+        name: impl Into<Text>,
+        jsdoc_comment: Option<String>,
+    ) -> Option<()> {
+        self.data
+            .exports
+            .insert(name.into(), Export::Own(OwnExport { jsdoc_comment }));
+
+        Some(())
+    }
+
+    fn import_from_module_source(&self, module_source: AnyJsModuleSource) -> Option<Import> {
+        let specifier = module_source
+            .as_js_module_source()?
+            .inner_string_text()
+            .ok()?;
+        Some(self.import_from_specifier(specifier.text()))
+    }
+
+    fn import_from_specifier(&self, specifier: &str) -> Import {
+        let resolved_path =
+            self.resolver
+                .resolve(self.directory, specifier)
+                .and_then(|resolution| {
+                    Utf8PathBuf::from_path_buf(resolution.into_path_buf())
+                        .map_err(|path| ResolveError::NotFound(path.to_string_lossy().to_string()))
+                });
+        Import { resolved_path }
+    }
+}
+
+fn get_name(binding_result: AnyJsBinding) -> Option<TokenText> {
+    let name = binding_result
+        .as_js_identifier_binding()?
+        .name_token()
+        .ok()?
+        .token_text_trimmed();
+    Some(name)
+}
+
+fn get_ts_name(binding_result: AnyTsIdentifierBinding) -> Option<TokenText> {
+    let name = binding_result
+        .as_ts_identifier_binding()?
+        .name_token()
+        .ok()?
+        .token_text_trimmed();
+    Some(name)
+}
+
+/// Normalizes the text in a JSDoc comment by stripping the opening and trailing
+/// markers, and trimming the leading asterisk from every line.
+fn normalize_jsdoc_comment(text: &str) -> String {
+    debug_assert!(text.starts_with("/**") && text.ends_with("*/"));
+
+    let mut result = text[3..text.len() - 2]
+        .lines()
+        .map(|line| {
+            let trimmed = line.trim();
+            trimmed.strip_prefix('*').map_or(trimmed, str::trim_start)
+        })
+        .fold(String::new(), |mut result, line| {
+            if !result.is_empty() {
+                result.push('\n');
+            }
+            result.push_str(line);
+            result
+        });
+
+    // Trim trailing newlines.
+    while result.ends_with('\n') {
```

Comment:
We should normalise newlines first (replace carriage returns with `\n`), then we can push our `\n` newlines, because comments might have `\r` and `\r\n` too

Reply:
We do normalise newlines implicitly by using `.lines()`, no?

Reply:
True!